### PR TITLE
Add upper bounds to reverse dependencies of Jane Street packages

### PR DESCRIPTION
**Pull request by xclerc**
Follow-up to https://github.com/ocaml/opam-repository/pull/13010, https://github.com/ocaml/opam-repository/pull/13016, https://github.com/ocaml/opam-repository/pull/13107, and https://github.com/ocaml/opam-repository/pull/13344.

### CODE REVIEW

**Comment by xclerc** [comment]
Once this PR is merged, we will actually release v0.12.

File: packages/biocaml/biocaml.0.9.0/opam [hunk 1]
```diff
@@ -41,8 +41,8 @@ depends: [
   "uri"
 ]
 depopts: [
-  "async"
-  "core"
+  "async" {< "v0.12"}
```
Comment (kit-ty-kate) [inline packages/biocaml/biocaml.0.9.0/opam]: This constraint should be in conflicts

**Comment by kit-ty-kate** [comment]
fixed. Should be ok now

**Comment by xclerc** [comment]
(Sorry, it looks like I forgot to update the script since the last PR...)

**Comment by mseri** [comment]
Looks fine

**Pull request merged**